commcare_connect/opportunity/api/views.py

```diff
@@ -30,3 +44,24 @@ def get(self, *args, **kwargs):
             "total_modules": total_modules.count(),
         }
         return Response(UserLearnProgressSerializer(ret).data)
+
+
+class UserVisitVerificationStatus(APIView):
+    serializer_class = UserVisitVerificationStatusSerializer
+    permission_classes = [IsAuthenticated]
+
+    def get(self, *args, **kwargs):
+        user_visits = UserVisit.objects.filter(user=self.request.user, opportunity=kwargs.get("pk"))
+        total_visits = user_visits.count()
+        daily_visits = user_visits.filter(visit_date__date=datetime.today().date()).count()
+        approved = user_visits.filter(status=VisitValidationStatus.approved).count()
+        pending = user_visits.filter(status=VisitValidationStatus.pending).count()
+        rejected = user_visits.filter(status=VisitValidationStatus.rejected).count()
+        ret = {
+            "approved": approved,
+            "pending": pending,
+            "rejected": rejected,
+            "total_visits": total_visits,
+            "daily_visits": daily_visits,
+        }
+        return Response(UserVisitVerificationStatusSerializer(ret).data)
```

Comment:
For simple data like this where you've already constructed the dict there's no need for a serializer:
```suggestion
        return Response(ret)
```